README.md

```diff
@@ -312,20 +316,63 @@ A few additional links to some typical early `Jupyter` topics:
 * [Closing running Jupyter notebook servers](https://github.com/jupyter/notebook/issues/2844)
 * [Checkpoints and autosave](https://groups.google.com/forum/#!topic/jupyter/DGCKE5fS4kQ)
 
-### Continuous Integration
+#### Command-line Interface (CLI)
+
+Python's ecosystem offers several ways to tackle command-line interfaces. The traditional standard method is to use
+the `argparse` module that is part of the standard library. This can be complemented by something like `argparsetree`
+for larger and more complex command-line applications.
+
+Here we have chosen to use [Click](https://click.palletsprojects.com/) instead, which allows us to define our CLI via
+decorated functions in a neat and compact way. Other potential alternatives could
+be [docopt](https://docopt.readthedocs.io/) or [Invoke](https://www.pyinvoke.org/).
+
+A nice comparison is
+available [here](https://realpython.com/comparing-python-command-line-parsing-libraries-argparse-docopt-click/).
+
+In order to install the package and **run the CLI commands**, you can follow the steps below.
+- set `PYTHONPATH` to point to this project. Then run:
+```bash
+python secretsanta/cli/cli.py --help
+```
+- try a specific command:
+```bash
+python secretsanta/cli/cli.py makedict --help
+python secretsanta/cli/cli.py makedict ".\validation\participants.json"
```

Comment:
Should we mention here that this is using the Windows slashes?

Reply:
Yea we need to be more accurate about Ubuntu vs Windows. For now I put just one `#Windows` comment in the section below, but we should have both where it matters and be clear about it otherwise.
Can you propose some additions to address this? I.e. Ubuntu way and proper distinction / comments?

Reply:
Btw, running the `makedict` command fails unless I manually create a `log_files` folder first. I guess we could just extend `setup_logging` to also create this folder if it is missing?

Reply:
fair point on the `log_files`, but move that to the postponed logging thing